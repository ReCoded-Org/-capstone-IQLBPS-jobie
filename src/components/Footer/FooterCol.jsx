import { Link } from "react-router-dom";

const FooterCol = ({ title, links }) => {
  return (
    <div>
      <p className="font-medium text-white opacity-30">{title}</p>
      <nav
        aria-label="Footer Nav"
        className="mt-3 flex flex-col space-y-1 text-sm small:text-base text-white"
      >
        {links.map((item, index) => {
          const key = index + 1;
          return (
            <Link key={key} className="hover:opacity-75" to={item.path}>
              {item.title}
            </Link>
          );
        })}

        {/* {links.map((item, index) => (
                <div>
                    <Link to={item.path}>{item.title}</Link>
                    
                </div>
            )
            } */}
        {/* </nav> */}
      </nav>
    </div>
  );
};

export default FooterCol;
