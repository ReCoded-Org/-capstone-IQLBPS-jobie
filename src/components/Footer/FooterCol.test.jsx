import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import FooterCol from "./FooterCol";

describe("Footer Component", () => {
  it("Footer Column when there are multiple items", () => {
    const domTree = renderer
      .create(
        <MemoryRouter>
          <FooterCol
            title="Title"
            links={[
              { title: "Home", path: "/" },
              { title: "About Us", path: "/About_Us" },
            ]}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
