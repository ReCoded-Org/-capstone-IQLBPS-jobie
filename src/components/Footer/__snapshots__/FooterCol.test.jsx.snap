// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Footer Column when there are multiple items 1`] = `
<div>
  <p
    className="font-medium text-white opacity-30"
  >
    Title
  </p>
  <nav
    aria-label="Footer Nav"
    className="mt-3 flex flex-col space-y-1 text-sm small:text-base text-white"
  >
    <a
      className="hover:opacity-75"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="hover:opacity-75"
      href="/About_Us"
      onClick={[Function]}
    >
      About Us
    </a>
  </nav>
</div>
`;
